lib/go-atscfg/headerrewritedotconfig.go

```diff
@@ -23,74 +23,300 @@ import (
 	"errors"
 	"fmt"
 	"math"
+	"net/url"
 	"regexp"
 	"strconv"
 	"strings"
 
 	"github.com/apache/trafficcontrol/lib/go-tc"
-	"github.com/apache/trafficcontrol/lib/go-util"
 )
 
 const HeaderRewritePrefix = "hdr_rw_"
+const HeaderRewriteMidPrefix = "hdr_rw_mid_"
 const ContentTypeHeaderRewriteDotConfig = ContentTypeTextASCII
 const LineCommentHeaderRewriteDotConfig = LineCommentHash
 
 const ServiceCategoryHeader = "CDN-SVC"
 
 const MaxOriginConnectionsNoMax = 0 // 0 indicates no limit on origin connections
 
+const HeaderRewriteFirstPrefix = HeaderRewritePrefix + "first_"
+const HeaderRewriteInnerPrefix = HeaderRewritePrefix + "inner_"
+const HeaderRewriteLastPrefix = HeaderRewritePrefix + "last_"
+
+func FirstHeaderRewriteConfigFileName(dsName string) string {
+	return HeaderRewriteFirstPrefix + dsName + ConfigSuffix
+}
+
+func InnerHeaderRewriteConfigFileName(dsName string) string {
+	return HeaderRewriteInnerPrefix + dsName + ConfigSuffix
+}
+
+func LastHeaderRewriteConfigFileName(dsName string) string {
+	return HeaderRewriteLastPrefix + dsName + ConfigSuffix
+}
+
+// MakeHeaderRewriteDotConfig makes the header rewrite file for
+// an Edge hdr_rw_ or Mid hdr_rw_mid_ or Topology hdr_rw_{first,inner,last} file,
```

Comment:
Is this comment correct? It looks like it should be `{first,inner,last}_hdr_rw_`, right? The order presented here suggests a potential conflict with a ds named `inner_`, for example.

Reply:
Yes, the comment is correct: https://github.com/apache/trafficcontrol/blob/0170bba93b4e8c3e9079b1d7f930d2c878de29e1/lib/go-atscfg/topologyheaderrewritedotconfig.go#L38

It matches the existing pattern. Yes, a DS named `inner` could cause problems, but that's true today for a DS named `mid` or `hdr_rw`.

For what it's worth, it won't cause problems once DeliveryServiceServers go away, and all DSes are using Topologies.

The only real way to fix it before that, would be to refactor and rename the DSS Edge file to be `hdr_rw_edge_*`, which would incur a non-change on every single Edge server when people upgrade.